core/src/main/java/com/arcrobotics/ftclib/hardware/motors/MotorGroup.java

```diff
@@ -40,6 +48,113 @@ public double get() {
         return group[0].get();
     }
 
+    /**
+     * @return All motor target speeds as a percentage of output
+     */
+    public List<Double> getSpeeds() {
+        return Arrays.stream(group)
+                .map(Motor::get)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public double getVelocity() {
+        return group[0].getCorrectedVelocity();
+    }
+
+    /**
+     * @return All current velocities of the motors in the group
+     */
+    public List<Double> getVelocities() {
+        return Arrays.stream(group)
+                .map(Motor::getCorrectedVelocity)
+                .collect(Collectors.toList());
+    }
+
+    @NotNull
+    @Override
+    public Iterator<Motor> iterator() {
+        return Arrays.asList(group).iterator();
+    }
+
+    @Override
+    public Encoder setDistancePerPulse(double distancePerPulse) {
+        Encoder leaderEncoder = group[0].setDistancePerPulse(distancePerPulse);
+        for (int i = 1; i < group.length; i++) {
+            group[i].setDistancePerPulse(distancePerPulse);
+        }
+        return leaderEncoder;
+    }
+
+    /**
+     * @return The position of every motor in the group in units of distance
+     * which is by default ticks
+     */
+    public List<Double> getPositions() {
+        return Arrays.stream(group)
+                .map(Motor::getDistance)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public void setRunMode(RunMode runmode) {
+        group[0].setRunMode(runmode);
+    }
+
+    @Override
+    public void setZeroPowerBehavior(ZeroPowerBehavior behavior) {
+        for (Motor x : group) {
+            x.setZeroPowerBehavior(behavior);
+        }
+    }
+
+    @Override
+    public void resetEncoder() {
+        for (Motor x : group) {
+            x.resetEncoder();
+        }
+    }
+
+    @Override
+    public void setPositionCoefficient(double kp) {
+        group[0].setPositionCoefficient(kp);
+    }
+
+    @Override
+    public boolean atTargetPosition() {
+        return group[0].atTargetPosition();
+    }
+
+    @Override
+    public void setTargetPosition(int target) {
+        group[0].setTargetPosition(target);
+    }
+
+    @Override
+    public void setTargetDistance(double target) {
+        group[0].setTargetDistance(target);
+    }
+
+    @Override
+    public void setPositionTolerance(double tolerance) {
+        group[0].setPositionTolerance(tolerance);
+    }
+
+    @Override
+    public void setVeloCoefficients(double kp, double ki, double kd) {
+        group[0].setVeloCoefficients(kp, ki, kd);
+    }
+
+    @Override
+    public void setFeedforwardCoefficients(double ks, double kv) {
+        group[0].setFeedforwardCoefficients(ks, kv);
+    }
+
+    @Override
+    public void setFeedforwardCoefficients(double ks, double kv, double ka) {
+        group[0].setFeedforwardCoefficients(ks, kv, ka);
```

Comment:
for all of these, are we only setting to one motor? shouldnt we do it to all in the group?

Reply:
This class now makes use of a leader/follower paradigm. The first motor in the group is the leader, so only it needs to have its configuration set. Everything else does not, since their speeds are set to whatever is output from the controllers in `group[0]`'s `set()` method.